consensus/signatures.go

```diff
@@ -12,220 +12,181 @@ var (
 )
 
 // Each input has a list of public keys and a required number of signatures.
-// InputSignatures keeps track of which public keys have been used and how many
+// inputSignatures keeps track of which public keys have been used and how many
 // more signatures are needed.
-type InputSignatures struct {
-	RemainingSignatures uint64
-	PossibleKeys        []SiaPublicKey
-	UsedKeys            map[uint64]struct{}
-	Index               int
+type inputSignatures struct {
+	remainingSignatures uint64
+	possibleKeys        []SiaPublicKey
+	usedKeys            map[uint64]struct{}
+	index               int
 }
 
-// sortedUnique checks that all of the elements in a []unit64 are sorted and
-// without repeates, and also checks that the largest element is less than or
-// equal to the biggest allowed element.
-func sortedUnique(elems []uint64, biggestAllowed int) (err error) {
+// sortedUnique checks that 'elems' is sorted, contains no repeats, and that no
+// element is an invalid index of 'elems'.
+func sortedUnique(elems []uint64) bool {
 	if len(elems) == 0 {
-		return
+		return true
 	}
 
 	biggest := elems[0]
 	for _, elem := range elems[1:] {
 		if elem <= biggest {
-			err = errors.New("covered fields sorting violation")
-			return
+			return false
 		}
 	}
-	if int(biggest) > biggestAllowed {
-		err = errors.New("covered fields indexing violation")
-		return
+	if biggest >= uint64(len(elems)) {
```

Comment:
I think I just implemented it wrong the first time around. This needs to be changed so that biggestAllowed == len (t.Item), and len(elems) == len(t.Sig.CF.Item)